x-pack/platform/plugins/private/reporting/server/features.ts

```diff
@@ -39,6 +40,41 @@ export function registerFeatures({ isServerless, features }: FeatureRegistration
         read: { disabled: true, savedObject: { all: [], read: [] }, ui: [] },
       },
     });
+  } else {
+    // Shell feature for self-managed environments, to be leveraged by a reserved privilege defined
+    // in ES. This grants access to reporting features in a legacy fashion.
+    features.registerKibanaFeature({
+      id: 'reportingLegacy',
+      name: i18n.translate('xpack.reporting.features.reportingLegacyFeatureName', {
+        defaultMessage: 'Reporting Legacy',
+      }),
+      category: DEFAULT_APP_CATEGORIES.management,
+      scope: [KibanaFeatureScope.Spaces, KibanaFeatureScope.Security],
+      app: [],
+      privileges: null,
+      reserved: {
+        description: i18n.translate(
+          'xpack.reporting.features.reportingLegacyFeatureReservedDescription',
+          {
+            defaultMessage:
+              'Reserved for use by the Reporting plugin. This feature is used to grant access to Reporting capabilities in a legacy manner.',
+          }
+        ),
+        privileges: [
+          {
+            id: 'reporting_user',
+            privilege: {
+              excludeFromBasePrivileges: true,
+              app: [],
+              catalogue: [],
+              savedObject: { all: [], read: [] },
+              api: ['generateReport', 'downloadCsv'], // NOTE: this was carried over from x-pack/platform/plugins/shared/features/server/oss_features.ts:850, but 'downloadCsv' is not used as an access control tag for any API endpoint. Remove it from here?
+              ui: ['generateReport'],
+            },
+          },
+        ],
+      },
+    });
```

Comment:
I can't say I fully know what each line of this is doing. I need really close review here